### ISSUE TITLE
Why output feature num not equal to vocab dict num

### ISSUE THREAD

**Issue by HeMuling** [kuleshov-group/caduceus#52]
When I'm playing with the pre-trained model `caduceus-ps_seqlen-131k_d_model-256_n_layer-16` from hugging fcae, I noticed the model to have structure of:
```
CaduceusForMaskedLM(
  (caduceus): Caduceus(
    (backbone): CaduceusMixerModel(
      (embeddings): CaduceusEmbeddings(
        (word_embeddings): RCPSEmbedding(
          (embedding): Embedding(16, 256)
        )
      )
      (layers): ModuleList(
        (0-15): 16 x RCPSMambaBlock(
          (mixer): RCPSWrapper(
            (submodule): BiMambaWrapper(
              (mamba_fwd): Mamba(
                (in_proj): Linear(in_features=256, out_features=1024, bias=False)
                (conv1d): Conv1d(512, 512, kernel_size=(4,), stride=(1,), padding=(3,), groups=512)
                (act): SiLU()
                (x_proj): Linear(in_features=512, out_features=48, bias=False)
                (dt_proj): Linear(in_features=16, out_features=512, bias=True)
                (out_proj): Linear(in_features=512, out_features=256, bias=False)
              )
              (mamba_rev): Mamba(
                (in_proj): Linear(in_features=256, out_features=1024, bias=False)
                (conv1d): Conv1d(512, 512, kernel_size=(4,), stride=(1,), padding=(3,), groups=512)
                (act): SiLU()
                (x_proj): Linear(in_features=512, out_features=48, bias=False)
                (dt_proj): Linear(in_features=16, out_features=512, bias=True)
                (out_proj): Linear(in_features=512, out_features=256, bias=False)
              )
            )
          )
          (norm): RMSNorm()
        )
      )
      (norm_f): RMSNorm()
    )
  )
  (lm_head): RCPSLMHead(
    (lm_head): Linear(in_features=256, out_features=16, bias=False)
  )
)
``` 
where the last layer (linear layer) has output feature number of 16.

However, when checking the vocab dict number, I noticed that the vocab dict has 12 tokens:
```
# get tokenizer vocab:
vocab = tokenizer.get_vocab()  
print(f"Vocabulary size: {len(vocab)}")  
for i, (word, index) in enumerate(vocab.items()):  
    print(f"{word}: {index}")
```
which gives:
```
Vocabulary size: 12
[CLS]: 0
[SEP]: 1
[BOS]: 2
[MASK]: 3
[PAD]: 4
[RESERVED]: 5
[UNK]: 6
A: 7
C: 8
G: 9
T: 10
N: 11
```
According to my understanding, the output feature number should be the same as the vocab dict number, such that, when I get logits by:
```
text = "GATGA" # an example
inputs = tokenizer(text, return_tensors="pt")
inputs.to("cuda")
with torch.no_grad():
    outputs = model(**inputs)
```
After getting logits, I can get the probability of what is the next token by:
```
last_token_logits = logits[:, -1, :]
last_token_probs = torch.nn.functional.softmax(last_token_logits, dim=-1)
```
and this probability should have shape of [batch_size, num of token], where the latter dim should be the probability of  what is the next token by, which should be 12.

However, the pre-trained model gives probability with shape of [batch_size, 16], and the probabilities of last 4 tokens (which shouldn't exit) are extremely low, e.g.:
```
tensor([[6.5201e-09, 5.7195e-09, 6.1669e-09, 1.3819e-08, 5.7324e-09, 5.9567e-09,
         5.3912e-09, 4.7203e-01, 1.0891e-01, 2.1776e-01, 2.0130e-01, 4.7939e-09,
         2.0721e-18, 2.1174e-18, 2.1093e-18, 2.0785e-18]], device='cuda:0')
```
Please let me know if this is a mistake or my incorrect understanding.

**Comment by yair-schiff** [comment]
There is a flag in the config called [`pad_vocab_size_multiple`](https://github.com/kuleshov-group/caduceus/blob/2593c807ebfc0aede05e7a991474b1a0e60fdc93/caduceus/configuration_caduceus.py#L24) which increases the embedding matrix / lm head dim to a multiple of the provided number, see [here](https://github.com/kuleshov-group/caduceus/blob/2593c807ebfc0aede05e7a991474b1a0e60fdc93/caduceus/modeling_caduceus.py#L352-L357). I used `8` for this parameters, so the original vocab size of 12 gets artificially increased to 16.
